@HOOptions/HOOptions.m

```diff
@@ -0,0 +1,322 @@
+                % wants to set.
+                if firstInputObj
+                    pvPairs = varargin(2:end);
+                else
+                    pvPairs = varargin;
+                end
+                
+                % Loop through each param-value pair and just try to set
+                % the option. When the option has been fully specified with
+                % the correct case, this is fast. The catch clause deals
+                % with partial matches or errors.
+                haveCreatedInputParser = false;
+                for i = 1:2:length(pvPairs)
+                    try
+                        obj.(pvPairs{i}) = pvPairs{i+1};
+                    catch ME %#ok
+                        
+                        % Create the input parser if we haven't already. We
+                        % do it here to avoid creating it if possible, as
+                        % it is slow to set up.
+                        if ~haveCreatedInputParser
+                            ip = inputParser;
+                            % Structures are currently not supported as
+                            % an input to optimoptions. Setting the
+                            % StructExpand property of the input parser to
+                            % false, forces the parser to treat the
+                            % structure as a single input and not a set of
+                            % param-value pairs.
+                            ip.StructExpand =  false;
+                            % Get list of option names
+                            allOptionNames = properties(obj);
+                            for j = 1:length(allOptionNames)
+                                % Just specify an empty default as we already have the
+                                % defaults in the options object.
+                                ip.addParameter(allOptionNames{j}, []);
+                            end
+                            haveCreatedInputParser = true;
+                        end
+                        
+                        % Get the p-v pair to parse.
+                        thisPair = pvPairs(i:min(i+1, length(pvPairs)));
+                        ip.parse(thisPair{:});
+                        
+                        % Determine the option that was specified in p-v pairs.
+                        % These options will now be matched even if only partially
+                        % specified (by 13a). Now set the specified value in the
+                        % options object.
+                        optionSet = setdiff(allOptionNames, ip.UsingDefaults);
+                        obj.(optionSet{1}) = ip.Results.(optionSet{1});
+                    end
+                end
+            end
+            
+            % Add required subclasses
+            obj.HOObservables = HOObservablesOptions();
+            
+        end
+        
+        function new = copy(this)
```

Comment:
we do not need a copy operator for non-handle classes. Please also delete this where copied it from (this is now the third time I have to complain about this in a PR :().